flight/Modules/Sensors/sensors.c

```diff
@@ -209,8 +212,18 @@ static void SensorsTask(void *parameters)
 		}
 
 		queue = PIOS_SENSORS_GetQueue(PIOS_SENSOR_BARO);
-		if (queue != NULL && PIOS_Queue_Receive(queue, &baro, 0) != false) {
-			update_baro(&baro);
+		if (queue != NULL) {
+			if (PIOS_Queue_Receive(queue, &baro, 0) != false) {
+				// we can use the timeval because it contains the current time stamp (PIOS_DELAY_GetRaw())
+				last_pios_sensor_update = timeval;
+				update_baro(&baro);
```

Comment:
Here you should probably add
```AlarmsClear(SYSTEMALARMS_ALARM_TEMPBARO);```
and wrap the code below in an else statement since there is no need to compare the time immediately after getting an update.